### PR TITLE
add multidex support

### PR DESCRIPTION
**Pull request by RonnyWu**
- update navigation version to '2.0.0-alpha3'
- update constraint layout version to '1.0.0-alpha09'
- feat:add multidex support
- style:replace IntelliJ IDEA default code style with kotlin coding conventions.

### CODE REVIEW

**Comment by RonnyWu** [comment]
I signed it!

**Comment by RonnyWu** [comment]
I signed it!

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -27,6 +27,7 @@
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
+        android:name="androidx.multidex.MultiDexApplication"
```
[the remark this answers is not in the document]
Comment (RonnyWu) [reply app/src/main/AndroidManifest.xml]: Of course, I think so.

**Comment by tiembo** [comment]
Hi @RonnyWu, thanks for the contribution.  Any particular reason for enabling MultiDex?

**Comment by andhie** [comment]
i do not see any reason to add MultiDex lib as the app havent hit the 64k limit (33935 methods as of writing).
other than that, there are a lot non related changes that is being stuffed together in this PR. it should be separated

**Comment by tiembo** [comment]
Hey @RonnyWu , thanks again for the PR.  While MultiDex is part of Jetpack, Sunflower is not a good candidate for MultiDex at this time, due to the reason @andhie mentioned (current method count is much less than the limit).

**Pull request closed without merging**